docs/language.md

```diff
@@ -6604,37 +6852,128 @@ TODO
 
 -->
 
-## Built-in Functions
+## Run-time Types
+
+Types can be represented at run-time.
+To create a type value, use the constructor function `Type<T>()`, which accepts the static type as a type argument.
+
+This is similar to e.g. `T.self` in Swift, `T::class` in Kotlin, and `T.class` in Java.
 
-### `panic`
+For example, to represent the type `Int` at run-time:
 
 ```cadence
-fun panic(_ message: String): Never
+let intType: Type = Type<Int>()
 ```
 
-Terminates the program unconditionally
-and reports a message which explains why the unrecoverable error occurred.
+This works for both built-in and user-defined types. For example, to get the type value for a resource:
 
-#### Example
+```cadence,file=runtime-type-collectible.cdc
+resource Collectible {}
 
-```cadence
-let optionalAccount: Account? = // ...
-let account = optionalAccount ?? panic("missing account")
+let collectibleType = Type<@Collectible>()
+
+// `collectibleType` has type `Type`
+```
+
+Type values are comparable.
+
+```cadence,file=runtime-type-comparison.cdc
+
+Type<Int>() == Type<Int>()
+
+Type<Int>() != Type<String>()
 ```
 
-### `assert`
+The function `fun isInstance(_ type: Type): Bool` can be used to check if a value has a certain type.
```

Comment:
Can you add more detail about how much `isInstance` checks the type? Will it know that an `@AnyResource` is a `@FlowToken.Vault` or if a `@FlowToken.Vault` is an `@AnyResource`?

Reply:
Good questions! Tried to improve it in 0b09123 and 13f6a52